command/crypto/jwt/sign.go

```diff
@@ -226,48 +225,28 @@ func signAction(ctx *cli.Context) error {
 		return errs.TooManyArguments(ctx)
 	}
 
-	alg := ctx.String("alg")
-	isSubtle := ctx.Bool("subtle")
-	kid := ctx.String("kid")
-
-	// Add parse options
-	var options []jose.Option
-	options = append(options, jose.WithUse("sig"))
-	if len(alg) > 0 {
-		options = append(options, jose.WithAlg(alg))
-	}
-	if len(kid) > 0 {
-		options = append(options, jose.WithKid(kid))
-	}
-	if isSubtle {
-		options = append(options, jose.WithSubtle(true))
-	}
-	if passwordFile := ctx.String("password-file"); len(passwordFile) > 0 {
-		options = append(options, jose.WithPasswordFile(passwordFile))
-	}
-
 	x5cCertFile, x5cKeyFile := ctx.String("x5c-cert"), ctx.String("x5c-key")
+	key := ctx.String("key")
+	jwks := ctx.String("jwks")
+	kid := ctx.String("kid")
 	var isX5C bool
-	if len(x5cCertFile) > 0 || len(x5cKeyFile) > 0 {
-		if len(x5cCertFile) == 0 {
-			return errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-cert")
-		}
+	if len(x5cCertFile) > 0 {
 		if len(x5cKeyFile) == 0 {
-			return errs.RequiredWithProvisionerTypeFlag(ctx, "X5C", "x5c-key")
+			if x5cKeyFile = key; len(x5cKeyFile) == 0 {
+				return errs.RequiredWithOrFlag(ctx, "x5c-cert", "key", "x5c-key")
+			}
 		}
-		isX5C = true
-	}
-
-	key := ctx.String("key")
-	jwks := ctx.String("jwks")
-	if isX5C {
-		if key != "" {
-			return errs.MutuallyExclusiveFlags(ctx, "x5c-key", "key")
+		if ctx.IsSet("jwk") {
+			return errs.MutuallyExclusiveFlags(ctx, "x5c-cert", "jwk")
 		}
-		if jwks != "" {
-			return errs.MutuallyExclusiveFlags(ctx, "x5c-key", "jwks")
+		if len(jwks) > 0 {
+			return errs.MutuallyExclusiveFlags(ctx, "x5c-cert", "jwks")
 		}
+		isX5C = true
 	} else {
+		if len(x5cKeyFile) > 0 {
+			return errs.RequiredWithFlag(ctx, "x5c-key", "x5c-cert")
```

Comment:
We can also use `errs.RequiredWithOrFlag`

Reply:
completely removed this clause. If you define `key` then `x5c-key` will also be defined because they are aliases.